### DISCUSSION TITLE
Change between beta 26 and 27 causes error in deployment

### DISCUSSION THREAD

**Discussion by brumble-dev** [deployphp/deployer#2787] [Help needed]
Hello,

I hope this is the right place to post this. When updating to the latest RC, I got an error in wirtable.php when deploying. I traced it back to beta 27. Beta 26 seems to be the last version where it worked for me.

The error I get when using beta 27 is:
`task deploy:writable`
`[staging] error in writable.php at line 115:`
`[staging] run cd /home/my/path/release && (setfacl -L -m u: "xxxx":rwX -m u:xxxx:rwX var)`
`[staging] err setfacl: Option -m: Invalid argument at character 3`
`[staging] exit code 2 (misuse of shell builtins)`

In RC 3, the message says the error occurs at line 120.

I have no idea what to change in my deployment to make it work. Could this be a bug or do I need to change something in my configuration? 

I am grateful for any help.

If you need more information to fix this problem, I will be happy to provide it.

![2021-11-26 13_29_15-web_ _var_www_html](https://user-images.githubusercontent.com/9021489/143589487-47d2fdd3-4ad6-40ee-8312-f1f703586296.png)

**Comment by antonmedv** [comment]
Please, show exactly `run cd /home/my/path/release && (setfacl -L -m u: "xxxx":rwX -m u:xxxx:rwX var)`, without any modification.

As error says something wrong on char 3.

**Reply by brumble-dev** [reply]
@antonmedv 
Thank you very much for your response :)

Here ist the exact command:
` run cd /home/weba2jr7y/my/path/release && (setfacl -L  -m u:"weba2jr7y":rwX -m u:weba2jr7y_4kxvgm:rwX var)`

**Reply by antonmedv** [reply]
Try to run it as is.

**Reply by brumble-dev** [reply]
@antonmedv

Thanks for pushing me in the right direction :)

When I run it on the server where I want to deploy to, I get the same error message:
`setfacl: Option -m: Invalid argument near character 3`

I am not an expert in server issues, but I tried to examine the problem a little further. So I educated myself on what this setfacl command does and when this error appears. It seems it occurs, when a user is not existing.

When I look at the non working command I see two users: weba2jr7y and weba2jr7y_4kxvgm, the first is the http user I defined in deploy.php the latter is the ssh/remote user. Therefore I thought it might make sense to run the command for both users separately.

`cd /home/weba2jr7y/html/my/path/release && (setfacl -L -m u:"weba2jr7y":rwX var)` works fine 
`cd /home/weba2jr7y/html/my/path/release && (setfacl -L -m u:"weba2jr7y_4kxvgm":rwX var)` throws the known error

When checking with `id weba2jr7y` / `id weba2jr7y_4kxvgm`, I can verify the second one does not exist as an ACL user. So that seems to be the configuration our hoster uses.

Given the fact it worked until beta 27, I guess the ssh/remote user was added to the command with this update. Why is that neccessary? And if it is, can there be an option to not use it or just use the http user?

I doubt that we will switch to another hoster because of this and also I don't think the hoster will change his configuration of manged servers.
